### PR TITLE
save review mutation

### PR DESCRIPTION
**Pull request by joyz7**
closes/addresses #49 

# Quick Overview of Changes

- created reviews router with save mutation

# Checklist

- [x] If any frontend changes were made, include a screenshot/demo in the overview
- [x] Checked all uses of changed component(s)/function(s)
- [x] If any changes were made to our backend services, at least one (happy path) unit test was added OR a `TODO` comment was added to create one.
- [x] Check that CI is passing.
- [x] If no (re-)reviews after 1-2 days, ping the web leads on the discord to remind them to review the PR.

@basokant @ArsalaanAli

### CODE REVIEW

File: src/server/db/schema.ts [hunk 1]
```diff
@@ -180,6 +180,7 @@ export const reviews = createTable(
     })
       .defaultNow()
       .notNull(),
+    completed: boolean("completed").default(false),
```
Comment (basokant) [inline src/server/db/schema.ts:183]: This is great! Thanks for adding this column, I forgot that we would need it LOL. Could you run `npm run db:generate` and npm run db:migrate` to add db migration scripts for adding this column?

Comment (basokant) [reply src/server/db/schema.ts:183]: before you run these, can you make `referral` default to false?

**Review by basokant — COMMENTED** [review]
Great work Joy! Sorry for the late review. Just a small comment, looks great! I don't know why I forgot that this autosaves the review just like for applications. Anyway, just a couple comments, should be easy to address :))

I'm going to create a ticket once this goes through for unit tests for this!

File: src/server/api/routers/reviews.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import { reviews } from "~/server/db/schema";
+import { db } from "~/server/db";
+import { reviewSaveSchema, reviewSubmitSchema } from "~/schemas/review";
+
+export const reviewsRouter = createTRPCRouter({
+  save: protectedProcedure
+    .input(reviewSaveSchema)
+    .mutation(async ({ input, ctx }) => {
+      try {
+        const userId = ctx.session.user.id;
```
Comment (basokant) [inline src/server/api/routers/reviews.ts:13]: Could you also check if the current user is an organizer? I forgot about this. Happy bday :))

File: src/server/db/schema.ts (hunk 1 above)
Comment (joyz7) [reply src/server/db/schema.ts:183]: Changed the referral default but I got this message when running migrate. Is there anything I gotta do?
image

File: src/server/api/routers/reviews.ts (hunk 2 above)
Comment (joyz7) [reply src/server/api/routers/reviews.ts:14]: Added and thank you :P

File: src/server/db/schema.ts (hunk 1 above)
Comment (basokant) [reply src/server/db/schema.ts:183]: nope! those are fine, I'm not sure why they always show up

**Review by basokant — APPROVED** [review]
lgtm!

**Pull request merged**